src/transformers/models/bert/modeling_bert.py

```diff
@@ -176,10 +176,19 @@ def __init__(self, config):
         # any TensorFlow checkpoint file
         self.LayerNorm = nn.LayerNorm(config.hidden_size, eps=config.layer_norm_eps)
         self.dropout = nn.Dropout(config.hidden_dropout_prob)
-
         # position_ids (1, len position emb) is contiguous in memory and exported when serialized
-        self.register_buffer("position_ids", torch.arange(config.max_position_embeddings).expand((1, -1)))
-        self.position_embedding_type = getattr(config, "position_embedding_type", "absolute")
+        try:
+            self.register_buffer(
+                "position_ids", torch.arange(config.max_position_embeddings).expand((1, -1)), persistent=False
+            )
+            self.position_embedding_type = getattr(config, "position_embedding_type", "absolute")
+            self.register_buffer(
+                "token_type_ids",
+                torch.zeros(self.position_ids.size(), dtype=torch.long, device=self.position_ids.device),
+                persistent=False,
+            )
+        except Exception:
+            logger.warning("Persistent flag for buffer registration is supported for Pytorch versions > 1.6.0")
```

Comment:
Some issues could be misidentified by having a bare exception here, how about checking for the version manually? It's done only once in the `__init__`, so it will only be called once.
How about something like the following:
https://github.com/huggingface/transformers/blob/cd56f3fe7eae4a53a9880e3f5e8f91877a78271c/src/transformers/activations.py#L45-L48

Reply:
Thanks @LysandreJik, it's been addressed.